awx/ui/src/screens/WorkflowApproval/shared/WorkflowApprovalControls.js

```diff
@@ -0,0 +1,157 @@
+import React from 'react';
+import { t } from '@lingui/macro';
+import {
+  Dropdown,
+  KebabToggle,
+  DropdownItem,
+  Tooltip,
+} from '@patternfly/react-core';
+import { useKebabifiedMenu } from 'contexts/Kebabified';
+import styled from 'styled-components';
+
+const Delete = styled(DropdownItem)`
+  &:first-child {
+    ${(props) => !props.isDisabled && `color:red`};
+  }
+`;
+
+function lackPermissions(item) {
+  return !item.can_approve_or_deny;
+}
+
+function isPending(item) {
+  return item.status === 'pending';
+}
+
+function WorkflowApprovalControls({
+  selected,
+  onHandleDeny,
+  onHandleCancel,
+  onOpenDeleteModal,
+  onToggleToolbarKebab,
+  isKebabOpen,
+}) {
+  const { isKebabified } = useKebabifiedMenu();
+
+  const hasSelectedItems = selected.length > 0;
+  const isApproveDenyOrCancelDisabled =
+    !hasSelectedItems ||
+    !selected.every(
+      (item) => item.status === 'pending' && item.can_approve_or_deny
+    );
+
+  const isDeleteDisabled =
+    !hasSelectedItems ||
+    selected.some(
+      (item) =>
+        item.status === 'pending' ||
+        !item.summary_fields.user_capabilities.delete
+    );
+  const isToggleDisabled = isDeleteDisabled && isApproveDenyOrCancelDisabled;
+
+  const renderTooltip = (option) => {
+    let content = option;
+    const itemsLackingPermissions = selected
+      .filter(lackPermissions)
+      .map((item) => item.name)
+      .join(', ');
+
+    const pendingItems = selected
+      .filter(isPending)
+      .map((item) => item.name)
+      .join(', ');
+
+    if (selected.some(lackPermissions)) {
+      content = t`You are unable to act on the following workflow approvals due to insufficient permission: ${itemsLackingPermissions}`;
```

Comment:
This content is being displayed even the user has permission to delete.

<img width="1478" alt="image" src="https://user-images.githubusercontent.com/9053044/159354349-79de266d-cf13-4f69-9de3-50471b75cdcb.png">